disneyApi/src/hooks/useFetchData.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+const useFetchData = (url) => {
+	const [data, setData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		const fetchData = async () => {
+			setIsLoading(true);
+			try {
+				const response = await fetch(url);
+				if (!response.ok) {
+					throw new Error('Wystąpił błąd, dane nie mogą zostać pobrane!');
+				}
+				const data = await response.json();
+				setData(data);
+			} catch (error) {
+				setError(error);
+			} finally {
+				setIsLoading(false);
+			}
+		};
+		fetchData();
+
+		return () => {
```

Comment:
The cleanup function in useEffect sets isLoading to false, which might not be necessary and could lead to unexpected behavior. Consider removing it. Better way is to set it inside if statement.